keras/src/ops/image.py

```diff
@@ -7,6 +7,78 @@
 from keras.src.ops.operation_utils import compute_conv_output_shape
 
 
+class PSNR(Operation):
+    def __init__(
+        self,
+        max_val,
+    ):
+        super().__init__()
+        self.max_val = max_val
+
+    def call(self, image1, image2):
+        return backend.image.psnr(
+            image1=image1,
+            image2=image2,
+            max_val=self.max_val,
+        )
+
+    def compute_output_spec(self, image1, image2):
+        if len(image1.shape) != len(image2.shape):
+            raise ValueError("Input images must have the same rank")
+
+        if len(image1.shape) not in (3, 4):
+            raise ValueError(
+                "Invalid image rank: expected rank 3 (single image) "
```

Comment:
There is nothing in the computation that relies on shapes. PSNR is something that is valid for any signal, not just images. So we should:

- Move this API to `nn` instead of `image`
- Remove all shape checks (except checking that the two input have the same shape)